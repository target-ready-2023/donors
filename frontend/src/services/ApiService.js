import axios from 'axios';

const DONOR_API_BASE_URL = "http://localhost:8080/api/donor";
const EMAIL_API_BASE_URL = "http://localhost:8080/api/email";
const ALL_DONOR_DETAIL_URL="http://localhost:8080/api/admin";
const ALL_TRANSACTION_URL="http://localhost:8080/api/admin";
const ALL_TRANSACTION_OF_SINGLE_DONOR="http://localhost:8080/api/admin"
export const addDonorInfo = (data) => {
    return axios.post(`${DONOR_API_BASE_URL}/addDonor`, data);
};
  
export const addTransactionInfo = (data) => {
    return axios.put(`${DONOR_API_BASE_URL}/addTransaction`, data);
};

export const getDonorDetailsById = (donorEmail) => {
    return axios.get(`${DONOR_API_BASE_URL}/findByEmail`, {
        params:{
            donorEmail,
        },
    });
};

export const getCertificate = (donorEmail, fiscalYear) => {
    return axios.get(`${EMAIL_API_BASE_URL}/send`, {
        params : {
            donorEmail, fiscalYear,
        },
    });
};

export const getallDonarDetail=()=>{
    return axios.get(`${ALL_DONOR_DETAIL_URL}/getAllDonors`)
}

export const getAllTransactionOfParticularDonor=(donorEmail)=>{
    return axios.get(`${ALL_TRANSACTION_OF_SINGLE_DONOR}/getTransactionsWithEmail`,{
        params:{
            donorEmail
        }
    })
}
export const getAllTransaction=()=>{
    return axios.get(`${ALL_TRANSACTION_URL}/getAllTransactions`)
}