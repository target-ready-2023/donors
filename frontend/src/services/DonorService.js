import axios from 'axios';

const DONOR_API_BASE_URL = "http://localhost:8080/api/donor";

class DonorService{
    getDonors(){
        return axios.get(DONOR_API_BASE_URL + '/getAllDonors');
    }

    createDonor(donor){
        return axios.post(DONOR_API_BASE_URL + '/addDonor', donor);
    }

    getDonorByEmail(donorEmail){
        return axios.get(DONOR_API_BASE_URL + '/findByEmail', donorEmail);
    }
}

export default new DonorService();