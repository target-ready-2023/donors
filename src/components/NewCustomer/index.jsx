import React, { useState } from "react";
import { TextField, Button, Stack, MenuItem } from "@mui/material";
import { Link } from "react-router-dom";
import swal from "sweetalert";

const options = [
  {
    label: "UPI",
    value: 1,
  },
  {
    label: "IMPS",
    value: 2,
  },
  {
    label: "DEBIT",
    value: 3,
  },
  {
    label: "CREDIT",
    value: 4,
  },
  {
    label: "NEFT",
    value: 5,
  },
];

const NewCustomer = () => {
  const [Name, setName] = useState("");
  const [Address, setAddress] = useState("");
  const [email, setEmail] = useState("");
  const [dateOfBirth, setDateOfBirth] = useState("");
  const [PAN, setPAN] = useState("");
  const [Amount, setAmount] = useState("");
  const [Transaction, setTransaction] = useState(options[0]);
  const [Date, setDate] = useState("");

  function handleSubmit(event) {
    event.preventDefault();
    setName("");
    setAddress("");
    setEmail("");
    setDateOfBirth("");
    setPAN("");
    setDate("");
    setAmount("");
    setTransaction(options[0]);
    console.log(
      Name,
      Address,
      email,
      dateOfBirth,
      PAN,
      Amount,
      Transaction,
      Date
    );
    swal({
      title: `Hello !!`,
      text: "You have successfully added !",
      icon: "success",
      buttons: {
        Ok: { text: "Great !" },
      },
    });
  }

  return (
    <React.Fragment>
      <div style={{ padding: "90px " }}>
        <div
          style={{
            margin: "0 auto",
            display: "flex",
            width: "650px",
            flexDirection: "column",
            alignItems: "center",
            justifyContent: "center",
            textAlign: "center",
            borderRadius: "5px",
            border: "3px solid grey",
          }}
        >
          <h2 style={{ color: "grey" }}>
            <center>New Customer</center>
          </h2>
          <hr style={{ width: "100%", borderTop: "2px solid grey" }}></hr>
          <form
            onSubmit={handleSubmit}
            action={<Link to="/existingCustomer" />}
          >
            <Stack
              spacing={8}
              direction="row"
              sx={{ marginTop: 4, marginBottom: 4 }}
            >
              <TextField
                type="text"
                variant="filled"
                label="Name"
                onChange={(e) => setName(e.target.value)}
                value={Name}
                fullWidth
                required
                InputProps={{ disableUnderline: true }}
              />
              <TextField
                type="text"
                variant="filled"
                label="Address"
                onChange={(e) => setAddress(e.target.value)}
                value={Address}
                fullWidth
                required
                InputProps={{ disableUnderline: true }}
              />
            </Stack>
            <Stack spacing={8} direction="row" sx={{ marginBottom: 0 }}>
              <TextField
                type="email"
                variant="filled"
                label="Email"
                onChange={(e) => setEmail(e.target.value)}
                value={email}
                fullWidth
                required
                sx={{
                  mb: 4,
                }}
                InputProps={{ disableUnderline: true }}
              />
              <TextField
                type="date"
                variant="filled"
                label="Date of Birth"
                onChange={(e) => setDateOfBirth(e.target.value)}
                value={dateOfBirth}
                fullWidth
                required
                sx={{
                  mb: 4,
                }}
                InputProps={{ disableUnderline: true }}
                InputLabelProps={{
                  shrink: true, disableUnderline: true 
                }}
              
              />
            </Stack>

            <Stack spacing={8} direction="row" sx={{ marginBottom: 0 }}>
              <TextField
                type="text"
                variant="filled"
                label="PAN"
                onChange={(e) => setPAN(e.target.value)}
                value={PAN}
                fullWidth
                required
                sx={{
                  mb: 4,
                }}
                InputProps={{ disableUnderline: true }}
              />

              <TextField
                type="number"
                variant="filled"
                label="Amount"
                onChange={(e) => setAmount(e.target.value)}
                value={Amount}
                fullWidth
                required
                sx={{
                  mb: 4,
                }}
                InputProps={{ disableUnderline: true }}
              />
            </Stack>
            <Stack spacing={8} direction="row" sx={{ marginBottom: 0 }}>
              <TextField
                select
                required="true"
                label="Transaction mode"
                fullWidth
                variant="filled"
                sx={{
                  mb: 4,
                }}
                InputProps={{ disableUnderline: true }}
              >
                {options.map((option) => (
                  <MenuItem key={option.value} value={option.value}>
                    {option.label}
                  </MenuItem>
                ))}
              </TextField>
              <TextField
                type="date"
                variant="filled"
                label="Date"
                onChange={(e) => setDate(e.target.value)}
                value={Date}
                fullWidth
                required
                sx={{
                  mb: 4,
                }}
                InputProps={{ disableUnderline: true }}
                InputLabelProps={{
                  shrink: true,
                }}
              />
            </Stack>
            <Button
              variant="outlined"
              style={{
                color: "white",
                marginLeft: "4%",
                marginBottom: "5%",
                backgroundColor: "#1167b1",
                paddingLeft: "30px",
                paddingRight: "30px",
              }}
              type="submit"
            >
              Submit
            </Button>
          </form>
        </div>
      </div>
    </React.Fragment>
  );
};

export default NewCustomer;
