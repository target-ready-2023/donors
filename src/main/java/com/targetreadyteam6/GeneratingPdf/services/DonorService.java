package com.targetreadyteam6.GeneratingPdf.services;


import com.targetreadyteam6.GeneratingPdf.entity.DonorProfile;
import com.targetreadyteam6.GeneratingPdf.repository.DonorRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class DonorService {
    @Autowired
    private DonorRepository donorRepository;

    public List<DonorProfile>  findAllDonor() {
        return donorRepository.findAll();
    }


    public DonorProfile findPersonById(Integer id) {
        Optional<DonorProfile> optionalDonor = donorRepository.findById(id);
        return optionalDonor.orElse(null);
    }

}
