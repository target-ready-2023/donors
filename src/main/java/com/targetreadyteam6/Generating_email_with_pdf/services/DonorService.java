package com.targetreadyteam6.Generating_email_with_pdf.services;



import com.targetreadyteam6.Generating_email_with_pdf.entities.DonorProfile;
import com.targetreadyteam6.Generating_email_with_pdf.repositories.DonorRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class DonorService {

    private final DonorRepository donorRepository;

    @Autowired
    public DonorService(DonorRepository donorRepository) {
        this.donorRepository = donorRepository;
    }


    public List<DonorProfile> findAllDonor() {
        return donorRepository.findAll();
    }


    public DonorProfile findPersonById(Integer id) {
        Optional<DonorProfile> optionalDonor = donorRepository.findById(id);
        return optionalDonor.orElse(null);
    }


}
