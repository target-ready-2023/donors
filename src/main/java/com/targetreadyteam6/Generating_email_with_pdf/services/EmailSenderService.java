
package com.targetreadyteam6.Generating_email_with_pdf.services;

import jakarta.mail.MessagingException;
import jakarta.mail.internet.MimeMessage;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.core.io.ByteArrayResource;
import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.mail.javamail.MimeMessageHelper;
import org.springframework.stereotype.Service;

@Service
public class EmailSenderService {
    @Autowired
    private JavaMailSender javaMailSender;
    public void sendMailWithAttachment(String toEmail,
                                       String subject,
                                       String body,String attachmentFileName,
                                       byte[] attachment) throws MessagingException {


        MimeMessage mimeMessage = javaMailSender.createMimeMessage();
        MimeMessageHelper mimeMessageHelper = new MimeMessageHelper(mimeMessage,true);
        mimeMessageHelper.setFrom("layabadireddi@gmail.com");
        mimeMessageHelper.setTo(toEmail);
        mimeMessageHelper.setText(body);
        mimeMessageHelper.setSubject(subject);

        ByteArrayResource byteArrayResource = new ByteArrayResource(attachment);
        mimeMessageHelper.addAttachment(attachmentFileName, byteArrayResource);
        javaMailSender.send(mimeMessage);
        System.out.println("Mail with attachment sent successfully...");

    }
}
