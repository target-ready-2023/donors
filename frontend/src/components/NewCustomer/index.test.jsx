import axios from 'axios';
import { addDonorInfo } from '../../services/ApiService';
import { render, screen, fireEvent } from "@testing-library/react";
import NewCustomer from "./index";
import { waitFor } from '@testing-library/react';
jest.mock('axios');

describe('New Donor', () => {

  const donorData = {donorName : "Akankshaaa",
    donorAddress : "Jaipur",
    donorEmail: "akankshafaujdar123@gmail.com",
    dateOfBirth : "2000-12-11",
    donorPan : "AHQF7122D",
    transactions : [{
      amount : '20000',
      transactionMode : 'UPI',
      transactionDate : '20-12-2019'
    }] };

  afterEach(() => {
    jest.clearAllMocks();
  });
//Testing of addDonorInfo 
  it('should add a new donor', async () => {
    const response = {status : 201 ,data:{ donorID : 1}};
    axios.post.mockResolvedValue(response);

    const result = await addDonorInfo(donorData);

    expect(axios.post).toHaveBeenCalledTimes(1);
    expect(axios.post).toHaveBeenCalledWith( `http://localhost:8080/api/donor/addDonor`,donorData);
    expect(result.data).toEqual(response.data); // Compare only the "data" property of the response
    expect(result.status).toEqual(response.status); // Compare only the "status" property of the response
 
  });
//Error Handling
  it('should handle errors', async () => {
    const errorResponse = { response: { status: 500, data: { error: 'Server Error' } } };
    axios.post.mockRejectedValue(errorResponse);

    await expect(addDonorInfo(donorData)).rejects.toEqual(errorResponse);
  });
//Rendering NewDonor
  test("renders the NewDonor component", () => {
    render(NewCustomer);
    const nameInput = screen.queryByLabelText("Name");
  waitFor(()=>expect(nameInput).toBeInTheDocument());
  });

  
});
