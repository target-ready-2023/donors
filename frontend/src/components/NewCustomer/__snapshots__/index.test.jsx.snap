// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Donor NewDonor component renders correctly 1`] = `<DocumentFragment />`;
